src/selectors.ts

```diff
@@ -0,0 +1,27 @@
+import { SelectionEntry, UIDisplayData } from '@aics/simularium-viewer';
+import { CheckboxState, VisibilitySelectionMap } from './constants';
+
+const { Checked } = CheckboxState;
+
+export const getNewHiddenAgents = (
+  uiDisplayData: UIDisplayData,
+  prevCheckboxState: CheckboxState
+): VisibilitySelectionMap => {
+  let payload: VisibilitySelectionMap = {};
+  if (prevCheckboxState === Checked) {
+    payload = uiDisplayData.reduce<VisibilitySelectionMap>((acc, item) => {
+      acc[item.name] = [];
+      return acc;
+    }, {});
+  }
+  return payload;
```

Comment:
I think I'm having a hard time following what this method is doing just from the variable naming and type names alone; it could maybe use a few extra comments. What does it mean for a `VisibilitySelectionMap` to be empty? 

When `prevCheckboxState === Checked`, that means you're hiding all the agents, so you return a dictionary initialized with all the agents...?

Reply:
I saw your comment above on Megan's question... this section could definitely use more documentation.

Reply:
Deleted this selector, and now using two memoized values (with comment in code) in the context file.